conan/cli/commands/workspace.py

```diff
@@ -101,6 +104,63 @@ def workspace_info(conan_api: ConanAPI, parser, subparser, *args):
     return {"info": conan_api.workspace.info()}
 
 
+@conan_subcommand()
+def workspace_build(conan_api: ConanAPI, parser, subparser, *args):
+    """
+    Build the current workspace, starting from the "products"
+    """
+    subparser.add_argument("path", nargs="?",
+                           help='Path to a package folder in the user workspace')
+    add_common_install_arguments(subparser)
+    add_lockfile_args(subparser)
+    args = parser.parse_args(*args)
+    # Basic collaborators: remotes, lockfile, profiles
+    remotes = conan_api.remotes.list(args.remote) if not args.no_remote else []
+    overrides = eval(args.lockfile_overrides) if args.lockfile_overrides else None
+    # The lockfile by default if not defined will be read from the root workspace folder
+    ws_folder = conan_api.workspace.folder()
+    lockfile = conan_api.lockfile.get_lockfile(lockfile=args.lockfile, conanfile_path=ws_folder,
+                                               cwd=None,
+                                               partial=args.lockfile_partial, overrides=overrides)
+    profile_host, profile_build = conan_api.profiles.get_profiles_from_args(args)
+    print_profiles(profile_host, profile_build)
+
+    build_mode = args.build or []
+    if "editable" not in build_mode:
+        ConanOutput().info("Adding '--build=editable' as build mode")
+        build_mode.append("editable")
+
+    if args.path:
+        products = [args.path]
+    else:  # all products
+        products = conan_api.workspace.products
```

Comment:
We should to check here that products is defined and throw and exception or warning message or it will throw an ugly error because its `None`